### DISCUSSION TITLE
Ruby v3.0.5 something wrong with LDFLAGS

### DISCUSSION THREAD

**Discussion by prpetten** [rbenv/ruby-build#2282] [Build failures] [answered]
rbenv - v 1.2.0
Processor - Apple M2 Max
Operating System - macOS Sonoma (14.0 build 23A344)
Command Line Tools - v 15 (2399)

````
$ rbenv install 3.0.5
To follow progress, use 'tail -f /var/folders/nx/86m1hyrj02vflg7ppczp1xfm0000gn/T/ruby-build.20231023211743.40161.log' or pass --verbose
Downloading openssl-1.1.1w.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/cf3098950cb4d853ad95c0841f1f9c6d3dc102dccfcacd521d93925208b76ac8
Installing openssl-1.1.1w...
Installed openssl-1.1.1w to /Users/paulpettengill/.rbenv/versions/3.0.5

Downloading ruby-3.0.5.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.5.tar.gz
Installing ruby-3.0.5...
ruby-build: using readline from homebrew
ruby-build: using libyaml from homebrew

BUILD FAILED (macOS 14.0 using ruby-build 20231014)

Inspect or clean up the working tree at /var/folders/nx/86m1hyrj02vflg7ppczp1xfm0000gn/T/ruby-build.20231023211743.40161.SBmpKP
Results logged to /var/folders/nx/86m1hyrj02vflg7ppczp1xfm0000gn/T/ruby-build.20231023211743.40161.log

Last 10 log lines:
use 'mkdir -p' as MKDIR_P
checking for dtrace... dtrace
checking for dot... no
checking for doxygen... no
checking for pkg-config... pkg-config
checking whether it is Android... no
checking for cd using physical directory... cd -P
checking whether CFLAGS is valid... yes
checking whether LDFLAGS is valid... no
configure: error: something wrong with LDFLAGS="-L/Users/paulpettengill/.rbenv/versions/3.0.5/lib "
````

**Comment by prpetten** [comment]
Not sure if this is enough information to get started, but happy to provide whatever logs might be helpful.  This is a brand new machine, trying to get it set up for work.

**Reply by prpetten** [reply]
I also looked at the similar issues, but didn't see one for this LDFLAGS issue

Running on bash, here is my `.bash_profile`
````
export PATH=/opt/homebrew/bin:$PATH
export LIBRARY_PATH=$LIBRARY_PATH:/usr/local/opt/openssl/lib/
export PATH="$HOME/.rbenv/bin:$PATH"

eval "$(rbenv init -)"
````

**Reply by monfresh** [reply, marked as answer]
Try removing the LIBRARY_PATH line from your .bash_profile.

Setting env vars like LIBRARY_PATH globally in your shell startup file is one of the most common reasons for Ruby installation failures, and can cause other tools to fail as well. These kinds of env vars should only be set if you know what you're doing, and should never be set globally in your shell startup file.

Also, the fact that the LIBRARY_PATH is pointing to `/usr/local` either means you blindly copied and pasted this, or that you installed Homebrew using Rosetta, neither of which I recommend. On a brand new Apple Silicon Mac, Hombrew should be installed in `/opt/homebrew`, and if you didn't install Homebrew in `/usr/local`, then pointing LIBRARY_PATH to a nonexistent directory can cause tools like Ruby to fail to install.

If removing LIBRARY_PATH doesn't work, try my [Ruby on Mac](https://www.rubyonmac.dev/) script. Aside from setting up a proper Ruby dev environment, it can save you hours when setting up a brand new machine with everything else you need like all your dev tools, Mac apps, fonts, macOS preferences, Git preferences, and GitHub repos that you need to clone.

**Reply by prpetten** [reply]
Awesome, that did it.  My homebrew was in `/opt/homebrew`

**Reply by jakezur1** [reply]
i hate my life. this worked im such a bot.

**Comment by lxysl** [comment]
Mine uses zsh. I revised my `~/.zshrc`, and removed `export PATH` items and it worked.

**Comment by marvellouschandan** [comment]
Since, I had Kafka installed, I had to comment below properties from my .zshrc file.

```
#export C_INCLUDE_PATH=/opt/homebrew/Cellar/librdkafka/2.3.0/include
#export LIBRARY_PATH=/opt/homebrew/Cellar/librdkafka/2.3.0/lib
```

After that run below commands on terminal
```bash
source .zshrc
LIBRARY_PATH=""
```

Finally install using below command
```bash
rvm install ruby-3.3.1 --with-openssl-dir=/opt/homebrew/opt/openssl@1.1
```

**Comment by khaldruga** [comment]
checking whether CFLAGS is valid... yes
checking whether LDFLAGS is valid... no
...
LDFLAGS might be set to any value.

when rvm downloads specific version (eg 2.7.2) in your $HOME./rvm/.... and executing ./configure,
it needs to successfully compile a C/C++ program to verify CFLAGS/LDFLAGS are usable.   you need to install a C compiler then compile a small C program (hello world maybe) and check if it successfully builds an executable.

retry running rvm again.

**Reply by jasonkarns** [reply]
I'm confused why we're talking about rvm here. Does rvm use ruby-build these days?

**Reply by khaldruga** [reply]
i thought that should be, but seeing logs during activation of versions it does the ff:
- download the source based on version requested
- pre/post configuration
- compilation
- deploy

anyways, im just giving a hint of what i did that works
